.github/workflows/release.yml

```diff
@@ -0,0 +1,92 @@
+#############################################################################
+# GitHub Action to run all phases of the release.
+#
+#############################################################################
+name: "Release"
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'Target version without (-SNAPSHOT)'
+        required: true
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
```

Comment:
@nrmancuso We run into a concurrency deadlock here. Either we have to remove it or use a new one.

Concurrency problem: 
https://github.com/stoyanK7/checkstyle/actions/runs/4254614589
```
Release: .github#L1
Canceling since a deadlock for concurrency group 'Release-refs/heads/master' was detected between 'top level workflow' and 'update-sources'
```

What concurrency group do you suggest we use?

Reply:
Remember that we pass `github` context to child workflows, this is probably causing the issue. Let's just remove all release workflows from the check at https://github.com/checkstyle/checkstyle/blob/d6d906980759598be045000fd32f3567d2b7c3d5/.ci/validation.sh#L1111

Reply:
> Remember that we pass github context to child workflows

I do remember that. My thought was - we can give `release.yml` some unique concurrency group so we can still take advantage of concurrency functionality. For example, `Caller-${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}`.

https://docs.github.com/en/actions/using-workflows/reusing-workflows#overview
> A workflow that uses another workflow is referred to as a "caller" workflow.

Reply:
Makes sense to me, let’s try it

Reply:
Works. All jobs are running.
https://github.com/stoyanK7/checkstyle/actions/runs/4274211452
![Screenshot 2023-02-26 at 10-06-29 Release · stoyanK7_checkstyle@738cec2](https://user-images.githubusercontent.com/23459549/221401619-0524f01a-5b63-490e-bb6a-c7e6f627b719.png)